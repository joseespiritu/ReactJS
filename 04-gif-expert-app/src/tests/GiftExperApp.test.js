import React from "react";
import { shallow } from "enzyme";
import GiftExperApp from "../GiftExperApp";

describe("Pruebas sobre el componente <GiftGridExperApp/>", () => {
  test("Debe de renderizar el componente correctamente", () => {
    const wrapper = shallow(<GiftExperApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar una lista de categorias", () => {
    const categories = ["One punch", "dragon ball"];
    const wrapper = shallow(<GiftExperApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
