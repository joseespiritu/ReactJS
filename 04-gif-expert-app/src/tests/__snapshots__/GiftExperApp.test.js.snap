// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GiftGridExperApp/> Debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GiftExperApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
  </ol>
</div>
`;

exports[`Pruebas sobre el componente <GiftGridExperApp/> Debe de renderizar el componente correctamente 1`] = `
<div>
  <h2>
    GiftExperApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
